ci: Adjust GitHub Actions workflow "update lexicon" to make PR instead of pushing to main branch

follow up for #36

The CI failed because of the branch protection rule where the workflow cannot directly push the `main` branch. Also, it's better to create a PR so that we can track the upstream lexicon update well.

## CI log
```shell
Run git config user.name "github-actions[bot]"
[main 9191114] chore: update lexicons.ts
 1 file changed, 18 insertions(+), 2 deletions(-)
remote: error: GH006: Protected branch update failed for refs/heads/main.        
remote: 
remote: - Changes must be made through a pull request.        
remote: 
remote: - 2 of 2 required status checks are expected.        
remote: 
remote: - You're not authorized to push to this branch. Visit https://docs.github.com/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches for more information.        
To https://github.com/tsky-dev/tsky
 ! [remote rejected] main -> main (protected branch hook declined)
error: failed to push some refs to 'https://github.com/tsky-dev/tsky'
Error: Process completed with exit code 1.
```

https://github.com/tsky-dev/tsky/actions/runs/12791155164/job/35658613653

--- a/.github/workflows/update-lexicons.yml
+++ b/.github/workflows/update-lexicons.yml
@@ -34,10 +34,19 @@ jobs:
         id: check-version-change
         run: echo "VERSION_CHANGED=$(pnpm run --silent --filter @tsky/lexicons check-version-change)" >> "$GITHUB_OUTPUT"
 
-      - name: 🔼 commit and push
+      - name: 🔼 Create PR to update lexicons
         if: ${{ steps.check-version-change.outputs.VERSION_CHANGED == 'yes' }}
+        env:
+          BRANCH_NAME: gh-actions-update-lexicons
         run: |
           git config user.name "github-actions[bot]"
           git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
-          git diff --quiet || (git add . && git commit -m "chore: update lexicons.ts")
-          git push
+          git diff --quiet || (
+            git switch --create $BRANCH_NAME &&
+            git add . &&
+            git commit -m "chore: update lexicons.ts"
+          )
+          git push -u origin $BRANCH_NAME
+          gh pr create \
+            --title 'chore: Update lexicons' \
+            --body 'This PR is made via GitHub Actions workflow ([update-lexicons.yml](https://github.com/tsky-dev/tsky/blob/main/.github/workflows/update-lexicons.yml)) automatically.'

